File: prime_bench.rb
def count_primeS(size)
	total = 0
	flags = Array.new(size, true)
	flags[0] = flags[1] = false
	for s in 2..size/2-1
		if flags[s]
			total = total + 1
			(s * 2).step(size - 1, s) {|s2|
				flags[s2] = false
			}
		end
	end
    for s in s..size-1
        if flags[s]
            total = total + 1
        end
    end
	return total
end

if ARGV.size() < 1 then
	puts "ruby #{$0} size [count]"
	exit 1
end
size = ARGV[0].to_i
if size < 1 then
	puts "ruby #{$0} size [count]"
	exit 1
end
count = (ARGV.size() < 2) ? 1 : ARGV[1].to_i
total = 0;
for c in 0..count-1
	total = count_prime(size)
end
puts "prime count = #{total}"

File: prime_bench.php
<?php
function count_prime($size) {
	$total = 0;
    $flags = array_pad(array(), $size, true);
	$flags[0] = $flags[1] = false;
    for ($s = 2; $s < $size / 2; ++$s) {
		if ($flags[$s]) {
			++$total;
            for ($s2 = $s * 2; $s2 < $size; $s2 += $s) {
                $flags[$s2] = false;
            }
        }
    }
	for (; $s < $size; ++$s) {
		if ($flags[$s]) {
			++$total;
		}
	}
	return $total;
}

if (count($argv) < 2) {
	print "php " . $argv[0] . " size [count]" . PHP_EOL;
	exit(1);
}
$size = $argv[1];
if ($size < 1) {
	print "php " . $argv[0] . " size [count]" . PHP_EOL;
	exit(1);
}
$count = (count($argv) < 3) ? 1 : $argv[2];
$total = 0;
for ($c = 0; $c < $count; ++$c) {
	$total = count_prime($size);
}
print "prime count = " . $total . PHP_EOL;
exit(0);

